storagenode/piecestore/endpoint.go

```diff
@@ -522,8 +523,12 @@ func (endpoint *Endpoint) Download(stream pb.Piecestore_DownloadServer) (err err
 			// TODO: add timeout here
 			message, err = stream.Recv()
 			if err != nil {
-				// err is io.EOF when uplink closed the connection, no need to return error
-				return ErrProtocol.Wrap(ignoreEOF(err))
+				if !errs2.IsCanceled(err) {
+					// err is io.EOF or canceled when uplink closed the connection, no need to return error
+					return ErrProtocol.Wrap(ignoreEOF(err))
+				}
+				endpoint.log.Debug("uplink canceled connection")
```

Comment:
Let's switch the logic around such that:

```
if errs2.IsCanceled(err) {
	endpoint.log.Debug("uplink canceled connection")
	return nil
}
return ErrProtocol.Wrap(ignoreEOF(err))
```

It will get rid of the negation.